File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/Prescription.cs
namespace CareFirstClinic.API.Models
{
    public enum PrescriptionStatus
    {
        Issued, // Ð? ðý?c bác s? kê ðõn nhýng chýa ðý?c nhà thu?c x? l?
        Dispensed, // Nhà thu?c ð? c?p phát thu?c cho b?nh nhân
        Cancelled, // Ðõn thu?c ð? b? h?y b?i bác s? ho?c nhà thu?c
        Expired // Ðõn thu?c ð? h?t h?n (thý?ng là sau m?t kho?ng th?i gian nh?t ð?nh k? t? ngày kê ðõn)
    }

    public class Prescription
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public Guid MedicalRecordId { get; set; }

        public PrescriptionStatus Status { get; set; } = PrescriptionStatus.Issued;

        public string? Notes { get; set; }
        public DateTime IssuedAt { get; set; } = DateTime.UtcNow;
        // Navigation
        public MedicalRecord? MedicalRecord { get; set; }
        public ICollection<PrescriptionDetail> Details { get; set; } = new List<PrescriptionDetail>();
    }
}

File: src/frontend/src/types/prescription.ts
// src/types/prescription.ts

export type PrescriptionStatus = 'Issued' | 'Dispensed' | 'Cancelled' | 'Expired';

export interface PrescriptionDetail {
  id: string;
  stockId: string;
  medicineName: string;
  medicineCode?: string;
  unit?: string;
  frequency: string;        // Tần suất: 2 lần/ngày
  durationDays: number;     // Số ngày dùng
  quantity: number;         // Số lượng xuất kho
  instructions?: string;    // Uống trước/sau ăn
  unitPrice: number;
  totalPrice: number;
}

export interface Prescription {
  id: string;
  medicalRecordId: string;
  status: PrescriptionStatus;
  notes?: string;
  issuedAt: string; // ISO date string
  details: PrescriptionDetail[];
}

export interface CreatePrescriptionDetailDTO {
  stockId: string;
  frequency: string;
  durationDays: number;
  quantity: number;
  instructions?: string;
}

export interface CreatePrescriptionDTO {
  medicalRecordId: string;
  notes?: string;
  details: CreatePrescriptionDetailDTO[];
}

export interface PrescriptionResponse {
  message: string;
  data: Prescription;
}
